rpcs3/Loader/mself.hpp

```diff
@@ -12,10 +12,10 @@ struct mself_header
 	be_t<u32> header_size; // ???
 	u8 reserved[0x28];
 
-	u32 get_count(u64 file_size)
+	u32 get_count(u64 file_size) const
 	{
 		// Fast sanity check
-		if (magic != "MSF"_u32 || ver != u32{1} || this->size != file_size) [[unlikely]]
+		if (magic != "MSF"_u32 || ver != u32{1} || (file_size - sizeof(mself_header)) / 0x40 < count || this->size != file_size) [[unlikely]]
```

Comment:
Please replace both occasions of 0x40 with some constant.
It's hard to tell what's the logic here if you use a "random" number

Reply:
fixed.